mmf/models/mmbt.py

```diff
@@ -504,9 +528,41 @@ def forward(self, sample_list: Dict[str, Tensor]):
 
 @registry.register_model("mmbt")
 class MMBT(BaseModel):
-    def __init__(self, config):
+    @dataclass
+    class Config(BaseModel.Config):
+        model: str = "mmbt"
+        # classification or pretraining
+        training_head_type: str = "pretraining"
+        bert_model_name: str = "bert-base-uncased"
+        direct_features_input: bool = False
+        freeze_text: bool = False
+        freeze_modal: bool = False
+        freeze_complete_base: bool = False
+        finetune_lr_multiplier: float = 1
+        # Dimension of the embedding finally returned by the modal encoder
+        modal_hidden_size: int = 2048
+        text_hidden_size: int = 768
+        num_labels: int = 2
+        # This actually is Union[ImageEncoderConfig, ImageFeatureEncoderConfig]
+        modal_encoder: Encoder.Config = ImageEncoder.Config(
+            type=ImageEncoderTypes.resnet152, params=ResNet152ImageEncoder.Config()
+        )
+        text_encoder: Encoder.Config = TextEncoder.Config(
+            type=TextEncoderTypes.transformer,
+            params=TransformerEncoder.Config(bert_model_name=II("bert_model_name")),
+        )
+        use_modal_start_token: bool = True
+        use_modal_end_token: bool = True
+        fused_feature_only: bool = False
+        output_dim: int = 768
+
+    def __init__(self, config: Union[DictConfig, Config], *args, **kwargs):
         super().__init__(config)
 
+    @classmethod
+    def from_params(cls, **kwargs):
+        return MMBT(OmegaConf.structured(cls.Config(**kwargs)))
```

Comment:
So for this we don't require the base model config yaml file anymore is that right? Should we remove it to avoid confusion? (I am worried as there are two sources of truth for default configurations for the MMBT model now)

Reply:
This is something I was discussing with Omry and he has a solution for. Anyway, removing this can end up as a breaking change. We can remove this altogether once we move to hydra.